test_all.py

```diff
@@ -62,51 +70,80 @@ def test_chatbot(capsys):
     sys.path.append("example_agents/chatbot")
     from example_agents.chatbot.main import ChatBot
     from example_agents.chatbot.env import MultiChatEnv
-    from agentos import run_agent
 
     env_generator = MultiChatEnv()
     # say something in the room for the agent to hear
     client_env = env_generator()
     client_env.reset()
     running_agent = run_agent(
-        ChatBot, env_generator, hz=10, max_iters=40, as_thread=True
+        ChatBot, env_generator, hz=100, max_iters=40, as_thread=True
     )
     while not running_agent.is_alive():
         pass
-    time.sleep(1)
+    time.sleep(0.1)
     response_txt, _, _, _ = client_env.step("one")
-    time.sleep(1)
+    time.sleep(0.1)
     response_txt, _, _, _ = client_env.step("two")
     assert response_txt == "one", "chatbot should repeat strings from memory"
-
     # TODO(andyk): also test CommandLineListener
 
 
-def test_rl_agents():
-    from agentos import run_agent
-    from example_agents.rl_agents.reinforce_agent import ReinforceAgent
-    from gym.envs.classic_control import CartPoleEnv
-
-    run_agent(ReinforceAgent, CartPoleEnv, max_iters=10)
+def run_agent_in_dir(
```

Comment:
Ah cool!  so pytest-virtualenv gives you a place to install your requirements during the test and it looks like the example agents all have their own independent requirements.  Nice way to eliminate the duplication!

What does the `-e` in the example agent requirement files do? Install the `agentos` library for the test?

Reply:
Yep, you got it.

And yes, `-e` installs a package from source (short for editable).

This way the example agents can run against the version of agentos that is in the project folder on which they are located, and thus actually exercise any code that a user might have changed/added.